### ISSUE TITLE
as_vector_with_length on vectorquantities

### ISSUE THREAD

**Issue by ipelupessy** [amusecode/amuse#341]
**Describe the bug**
I not sure this makes sense currently:
```
>>> a=[1,2,3] | units.m
>>> print a.as_vector_with_length(5)
[[1, 2, 3]] m
```

**Expected behavior**
I don't know if length =1 then:
```
>>> a=[1] | units.m
>>> print a.as_vector_with_length(5)
[1,1, 1,1,1] m
```
and otherwise exception
or:
```
>>> a=[1,2,3] | units.m
>>> print a.as_vector_with_length(2)
[[1,2,3].[1,2,3]] m
```
(ie the increase in rank + some extra code..)

**Comment by arjenve** [comment]
Yes, I think you are right, it does not make sense to have a as_vector_with_length on a vector quantity like this. 

I would go for the exception. 

I see it is never used on vector quantities at the moment in the source. So yes, if length == 1 then extend else error...
